src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+
+import core.basesyntax.RegistrationUserException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void addNewUser_Ok() {
+        User expected = new User();
+        expected.setLogin("admin98@gmail.com");
+        expected.setPassword("Admin123");
+        expected.setAge(23);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void addNullUser_NotOk() {
+        assertThrows(RegistrationUserException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void addNullPassword_NotOk() {
+        User expected = new User();
+        expected.setLogin("admin@gmail.com");
+        expected.setPassword(null);
+        expected.setAge(23);
+        assertThrows(RegistrationUserException.class,
+                () -> registrationService.register(expected),
+                "Password cannot be empty");
+
+    }
+
+    @Test
+    void addShortPassword_NotOk() {
+        User expected = new User();
+        expected.setLogin("admin@gmail.com");
+        expected.setPassword("Admi");
+        expected.setAge(23);
+        assertThrows(RegistrationUserException.class, () ->
+                registrationService.register(expected),
+                "Password cannot be less than 6 characters");
+    }
+
+    @Test
+    void addNullLogin_NotOk() {
+        User expected = new User();
+        expected.setLogin(null);
+        expected.setPassword("Admin123");
+        expected.setAge(23);
+        assertThrows(RegistrationUserException.class,
+                () -> registrationService.register(expected),
+                "Login cant be null");
+    }
+
+    @Test
+    void addShortLogin_NotOk() {
+        User expected = new User();
+        expected.setLogin("admin");
+        expected.setPassword("Admin123");
+        expected.setAge(23);
+        assertThrows(RegistrationUserException.class, () ->
+                registrationService.register(expected),
+                "Login cant be less than 6 characters");
+    }
+
+    @Test
+    void invalidAge_NotOk() {
+        User expected = new User();
+        expected.setLogin("admin@gmail.com");
+        expected.setPassword("Admin123");
+        expected.setAge(17);
+        assertThrows(RegistrationUserException.class,
+                () -> registrationService.register(expected),
+                "Age must be at least 18 years old.");
+    }
+
+    @Test
+    void addRegisteredUser_NotOk() {
+        User expected = new User();
+        expected.setLogin("admin@gmail.com");
+        expected.setPassword("Admin123");
+        expected.setAge(23);
+        registrationService.register(expected);
```

Comment:
you can't use the method that you test for data preparation. Add the user to the storage directly